beamer/tests/health/test_check.py

```diff
@@ -0,0 +1,313 @@
+    ADDRESS1,
+    BLOCK_NUMBER,
+    CLAIM_ID,
+    CLAIMER_STAKE,
+    REQUEST_ID,
+)
+from beamer.tests.agent.utils import make_address
+
+TARGET_CHAIN_ID = ChainId(42161)
+SOURCE_CHAIN_ID = ChainId(10)
+TARGET_TOKEN_ADDRESS = make_address()
+SOURCE_TOKEN_ADDRESS = make_address()
+AGENT_ADDRESS = make_address()
+EVIL_AGENT_ADDRESS = make_address()
+
+request = RequestCreated(
+    request_id=REQUEST_ID,
+    target_chain_id=TARGET_CHAIN_ID,
+    source_token_address=SOURCE_TOKEN_ADDRESS,
+    target_token_address=TARGET_TOKEN_ADDRESS,
+    source_address=ADDRESS1,
+    target_address=ADDRESS1,
+    amount=TokenAmount(123),
+    nonce=Nonce(123),
+    valid_until=Termination(123),
+    block_number=BLOCK_NUMBER,
+    tx_hash=HexBytes(b"1"),
+    chain_id=SOURCE_CHAIN_ID,
+)
+
+fill = RequestFilled(
+    request_id=REQUEST_ID,
+    chain_id=TARGET_CHAIN_ID,
+    fill_id=FillId(b"1"),
+    source_chain_id=SOURCE_CHAIN_ID,
+    target_token_address=TARGET_TOKEN_ADDRESS,
+    filler=AGENT_ADDRESS,
+    amount=TokenAmount(100),
+    block_number=BLOCK_NUMBER,
+    tx_hash=HexBytes(b"1"),
+)
+
+claim = ClaimMade(
+    request_id=REQUEST_ID,
+    claim_id=CLAIM_ID,
+    fill_id=FillId(b"1"),
+    claimer=AGENT_ADDRESS,
+    claimer_stake=CLAIMER_STAKE,
+    chain_id=TARGET_CHAIN_ID,
+    last_challenger=ZERO_ADDRESS,
+    challenger_stake_total=Wei(0),
+    termination=Termination(100),
+    block_number=BLOCK_NUMBER,
+    tx_hash=HexBytes(b"1"),
+)
+
+
+def test_create_transfers_object():
+    events_grouped_by_chain_id: ChainEventMap = {
+        SOURCE_CHAIN_ID: [request, claim],
+        TARGET_CHAIN_ID: [fill],
+    }
+
+    transfers = create_transfers_object(events_grouped_by_chain_id)
+
+    assert transfers[REQUEST_ID.hex()]["created"] == request
+    assert transfers[REQUEST_ID.hex()]["filled"] == fill
+    assert transfers[REQUEST_ID.hex()]["claimed"][CLAIM_ID] == [claim]
+
+
+def test_check_for_request_created(ctx):
+    transfer: Transfer = {
+        "created": request,
+    }
+
+    check_if_request_created(transfer, ctx)
+
+    assert ctx.stats.requests == 1
+
+
+def test_check_if_request_has_fill_with_no_request_created(ctx):
+    transfer: Transfer = {
+        "filled": fill,
+    }
+
+    check_if_request_has_fill(transfer, ctx)
+
+    assert ctx.stats.fills == 0
+    assert ctx.stats.expired_requests == 0
+
+
+def test_check_if_request_has_fill_with_request_created(ctx):
+    transfer: Transfer = {
+        "created": request,
+        "filled": fill,
+    }
+
+    check_if_request_has_fill(transfer, ctx)
+    assert ctx.stats.fills == 1
+    assert ctx.stats.expired_requests == 0
+
+
+def test_check_if_request_has_fill_with_no_fill_and_request_created_not_expired(ctx):
+    transfer: Transfer = {
+        "created": dataclasses.replace(request, valid_until=Termination(int(time.time() + 1000))),
+    }
+
+    check_if_request_has_fill(transfer, ctx)
+
+    assert ctx.stats.fills == 0
+    assert ctx.stats.expired_requests == 0
```

Comment:
I'm going to make a general comment regarding testing here. I find this kind of granular testing a bad approach in terms of maintainability and developer productivity facilitation. Such "unit testing" does not make us more confident in our code than alternative approaches (outlined below). It just forces exposing all the internal check methods and, instead of helping future changes, makes it hard to modify anything because of inevitable test breakages. For an excellent overview of what is wrong with this, and general with modern-day testing, I can recommend the following talk by Ian Cooper: https://www.youtube.com/watch?v=EZ05e7EMOLM.

An alternative approach:
- have a well-defined, public entry point e.g. `analyze_transfers` -- this would be your "unit" of testing
- feed the entry point with various input -- these would be different test cases/functions
- check the resulting output to make sure it is correct

In this way, you do not need to care about the particular implementation of the checks, such as function names, arguments etc. What you care about is _functional correctness_ i.e. that your checks perform the job they were designed to do. This achieves the same effect, however it almost completely decouples the implementation from functionality and allows for far easier changes and refactorings in the future.